cirq-ft/cirq_ft/algos/mean_estimation/complex_phase_oracle.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence
+
+import attr
+import cirq
+from cirq._compat import cached_property
+from cirq_ft import infra
+from cirq_ft.algos import select_and_prepare
+from cirq_ft.algos.mean_estimation import arctan
+
+
+@attr.frozen
+class ComplexPhaseOracle(infra.GateWithRegisters):
+    r"""Applies $ROT_{y}|l>|garbage_{l}> = exp(i * -2arctan{y_{l}})|l>|garbage_{l}>$.
+
+    TODO: This currently assumes that the random variable `y_{l}` only takes integer
```

Comment:
Should we create an issue for this TODO and link it here?

Reply:
done